File: test_get_json_news.php
<?php

set_include_path("./service_spec");

require 'vendor/autoload.php';
include("GPBMetadata/Aigents.php");
include("Aigents/AigentsNewsFeedClient.php");
include("Aigents/Channel.php");
include("Aigents/Feed.php");
include("Aigents/Feeds.php");
include("Aigents/JSONFeed.php");
include("Aigents/Response.php");
include("Aigents/Site.php");
include("Aigents/Topic.php");
include("Aigents/UserInfo.php");

$cl = new Aigents\AigentsNewsFeedClient('localhost:9999',
    ['credentials' => Grpc\ChannelCredentials::createInsecure(),]);

function grpc_response($resp)
{
    if ($resp->code == Grpc\STATUS_OK) {
        echo "Call successful!\n";
        return true;
    }
    else {
        echo "Call NOT successful!\n";
        var_dump($resp);
        return false;
    }
}

function user_login($email, $sec_q, $sec_a)
{
    global $cl;
    echo "Calling UserLogin Service\n";
    $usri = new Aigents\UserInfo();
    $usri->setEmail($email);
    $usri->setSecretQuestion($sec_q);
    $usri->setSecretAnswer($sec_a);
    list($resp, $stat) = $cl->userLogin($usri)->wait();
    if (grpc_response($stat)) {
        $result = $resp->getText();
        echo "Result => Login: $result\n";
    }
}

function fetch_json_feed($channel)
{
    global $cl;
    echo "Calling reqJSON Service\n";
    $chn = new Aigents\Channel();
    $chn->setName($channel);
    list($resp, $stat) = $cl->reqJSON($chn)->wait();
    if (grpc_response($stat)) {
        $result = $resp->getNewsFeed();
        echo "Result => Request JSON: $result\n";
    }
}

function add_topic($label, $pattern)
{
    global $cl;
    echo "Calling addTopic\n";
    $tpc = new Aigents\Topic();
    $tpc->setLabel($label);
    $tpc->setPattern($pattern);
    list($resp, $stat) = $cl->addTopic($tpc)->wait();
    if (grpc_response($stat)) {
        $result = $resp->getText();
        echo "Result => Add Topic: $result\n";
    }
}

function add_sites($site)
{
    global $cl;
    echo "Calling addSite\n";
    $st = new Aigents\Sites();
    $st->setSite($site);
    list($resp, $stat) = $cl->addSite($st)->wait();
    if (grpc_response($stat)) {
        $result = $resp->getText();
        echo "Result => Add Site: $result\n";
    }
}
//user_login("<email>", "<secrect_question>", "<secret_answer>");
fetch_json_feed("satiretech");

?>

File: test_get_json_news.py
import sys
import grpc

sys.path.append("./service_spec")
import aigents_pb2 as pb2
import aigents_pb2_grpc as pb2_grpc

def user_signup(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    ui = pb2.UserInfo()
    ui.name = "TestName"
    ui.surname = "TestSurname"
    ui.email = "test@testemail.io"
    ui.secret_question = "testQuestion"
    ui.secret_answer = "testAnswer"
    resp = stub.userSignup(ui)
    print("Called userSignup ---> ", resp)
    assert resp.text == "OK"

def user_login(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    ui = pb2.UserInfo()
    ui.email = "test@testemail.io"
    ui.secret_question = "testQuestion"
    ui.secret_answer = "testAnswer"
    resp = stub.userLogin(ui)
    print("Called userLogin ---> ", resp)
    assert resp.text == "OK"

def add_topic(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    t = pb2.Topic()
    t.pattern = '{test pattern [for topic]}'
    resp = stub.addTopic(t)
    print("Called addTopic ---->" , resp)
    assert resp.text == "OK"

def add_topics(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    t0 = pb2.Topic(pattern='{test pattern one [for topic one]}')
    t1 = pb2.Topic(pattern='{test pattern two [for topic two]}')
    ts = pb2.Topics()
    ts.topics.extend([t0, t1])
    resp = stub.addTopics(ts)
    print("Called addTopics ---->" , resp)
    assert resp.text == "OK"

def add_site(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    s = pb2.Site()
    s.url = 'https://test-url.io'
    resp = stub.addSite(s)
    print("Called addSite --->" , resp)
    assert resp.text == "OK"

def add_sites(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    s0 = pb2.Site(url="http://test-url-one.com")
    s1 = pb2.Site(url="http://test-url-two.com")
    ss = pb2.Sites()
    ss.sites.extend([s0, s1])
    resp = stub.addSites(ss)
    print("Called addSites --->" , resp)
    assert resp.text == "OK"

def remove_topic(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    t = pb2.Topic()
    t.pattern = '{non existant test pattern [for topic]}'
    resp = stub.rmTopic(t)
    print("Called rmTopic to fail ---->" , resp)
    assert resp.text == "FAIL"
    t = pb2.Topic()
    t.pattern = '{test pattern [for topic]}'
    resp = stub.rmTopic(t)
    print("Called rmTopic ---->" , resp)
    assert resp.text == "OK"

def remove_topics(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    t0 = pb2.Topic(pattern='{test pattern one [for topic one]}')
    t1 = pb2.Topic(pattern='{test pattern two [for topic two]}')
    ts = pb2.Topics()
    ts.topics.extend([t0, t1])
    resp = stub.rmTopics(ts)
    print("Called rmTopics ---->" , resp)
    assert resp.text == "OK"

def remove_site(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    s = pb2.Site()
    s.url = 'https://test-url.io'
    resp = stub.rmSite(s)
    print("Called rmSite ---->" , resp)
    assert resp.text == "OK"

def remove_sites(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    s0 = pb2.Site(url="http://test-url-one.com")
    s1 = pb2.Site(url="http://test-url-two.com")
    ss = pb2.Sites()
    ss.sites.extend([s0, s1])
    resp = stub.rmSites(ss)
    print("Called rmSites --->" , resp)
    assert resp.text == "OK"

def create_news(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    ni = pb2.NewsItem()
    ni.title = "Test Article Title"
    ni.date = "2020-01-30"
    ni.url = "https://aigents.icog-labs.com"
    resp = stub.createNews(ni)
    print("Called createNews ---->" , resp)
    assert resp.text == "OK"

def vote_news(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    ni = pb2.NewsItem()
    ni.title = "Test Article Title"
    ni.date = "2020-01-30"
    ni.url = "https://aigents.icog-labs.com"
    ni.vote = 0
    resp = stub.voteNews(ni)
    print("Called voteNews ---->" , resp)
    assert resp.text == "OK"

def remove_news(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    ni = pb2.NewsItem()
    ni.title = "Test Article Title"
    ni.date = "2020-01-30"
    ni.url = "https://aigents.icog-labs.com"
    resp = stub.rmNews(ni)
    print("Called rmNews ---->" , resp)
    assert resp.text == "OK"

def make_friend(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    pa = pb2.PeerAction()
    pa.email = "dagim@icog-labs.com"
    pa.action = True
    resp = stub.mkFriend(pa)
    print("Called mkFriend ---->" , resp)
    assert resp.text == "OK"

def share_to_peer(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    pa = pb2.PeerAction()
    pa.email = "dagim@icog-labs.com"
    pa.action = True
    resp = stub.sharePeer(pa)
    print("Called sharePeer ---->" , resp)
    assert resp.text == "OK"

def receive_from_peer(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    pa = pb2.PeerAction()
    pa.email = "dagim@icog-labs.com"
    pa.action = True
    resp = stub.receivePeer(pa)
    print("Called receivePeer ---->" , resp)
    assert resp.text == "OK"

def get_news_feed(channel):
    stub = pb2_grpc.AigentsNewsFeedStub(channel)
    ch = pb2.Channel()
    ch.name = "satiretech"
    response = stub.reqJSON(ch)
    print(response)
    return response

with grpc.insecure_channel('localhost:9999') as channel:
    user_signup(channel)
    user_login(channel)
    add_topic(channel)
    add_topics(channel)
    add_site(channel)
    add_sites(channel)
    remove_topic(channel)
    remove_topics(channel)
    remove_site(channel)
    remove_sites(channel)
    make_friend(channel)
    share_to_peer(channel)
    receive_from_peer(channel)
    get_news_feed(channel)
